### DISCUSSION TITLE
Pass a Blade.php directly?

### DISCUSSION THREAD

**Discussion by kieranbarlow** [spatie/browsershot#661] [Ideas]
Hello,

This may be a daft question but would it be possible to pass a blade.php directly to the Browsershot instance and create a PDF? 

This would remove the need for me to install DomPDF?

**Comment by musik92** [comment]
@kieranbarlow what do you need DomPDF for?

**Reply by kieranbarlow** [reply]
Producing a PDF from a view, then emailed. Say a certificate.

**Reply by musik92** [reply]
Isn't it what you use browsershot for?

https://spatie.be/docs/browsershot/v2/usage/creating-pdfs

The point is to replace outdated solutions like domPDF.

**Comment by vincemammoliti** [comment]
You can render a blade.php view to get the html and then pass this to the Browsershot instance.

```
$html = view('example')->render();

Browsershot::html($html)->save('example.pdf');
```
